File: front/server/middleware/cors.ts
// server/middleware/cors.ts
import { defineEventHandler, setHeader } from "h3";

export default defineEventHandler((event) => {
  const allowedOrigins = [
    "http://localhost:3000",
    "https://your-domain.com",
  ] as const;
  const origin = event.node.req.headers.origin as string | undefined;

  if (origin && allowedOrigins.includes(origin)) {
    setHeader(event, "Access-Control-Allow-Origin", origin);
  }

  setHeader(
    event,
    "Access-Control-Allow-Methods",
    "GET, POST, PUT, DELETE, OPTIONS"
  );
  setHeader(
    event,
    "Access-Control-Allow-Headers",
    "Content-Type, Authorization"
  );
  setHeader(event, "Access-Control-Allow-Credentials", "true");

  if (event.method === "OPTIONS") {
    event.node.res.statusCode = 204;
    event.node.res.end();
    return;
  }
});

File: backend/config/initializers/cors.rb
# frozen_string_literal: true

Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'http://localhost:8001'

    resource '*',
             headers: :any,
             expose: %w[access-token expiry token-type uid client],
             methods: %i[get post put patch delete options head],
             credentials: true
  end
end
